### PR TITLE
Add tests for migration and persisted state

### PR DESCRIPTION
**Pull request by estrattonbailey**
Started to write tests for all persisted state APIs but got bogged down by handling the broadcast channel mocks. I can come back to that in a separate PR.

### CODE REVIEW

File: src/state/persisted/legacy.ts [hunk 1]
```diff
@@ -7,15 +7,15 @@ import {write, read} from '#/state/persisted/store'
...
+export type LegacySchema = {
   shell: {
     colorMode: 'system' | 'light' | 'dark'
   }
   session: {
     data: {
       service: string
       did: `did:plc:${string}`
-    }
+    } | null
```
Comment (estrattonbailey) [inline src/state/persisted/legacy.ts:18]: Small type update. Access below was using optional chaining, but this is more correct.

File: src/state/persisted/legacy.ts [hunk 2]
```diff
@@ -61,7 +61,7 @@ type LegacySchema = {
     copiedInvites: string[]
   }
   mutedThreads: {uris: string[]}
-  reminders: {lastEmailConfirm: string}
+  reminders: {lastEmailConfirm?: string}
```
Comment (estrattonbailey) [inline src/state/persisted/legacy.ts:64]: Small type update. Access below was using optional chaining, but this is more correct.

**Review by pfrazee — APPROVED** [review]
good stuff

**Comment by pfrazee** [comment]
@estrattonbailey Getting some weird sentry errors in the CI

**Pull request merged**